### ISSUE TITLE
Improve error reporting when an Ogg audio file fails to import

### ISSUE THREAD

**Issue by 31** [godotengine/godot#46486]
I tried to load `hello.ogg` into Godot, but it's actually an MP3 file. 😄 So, this issue now tracks making it a little more obvious in the UI what's going on. Original issue text below:

---

## Original issue

**Godot version:**
Reproduces in 3.2.3 and 3.2.4-rc3

**OS/device including version:**
Windows 10 x64

**Issue description:**
Adding this ogg file (attached in repro steps) to a project gives me some error logs in the Godot console:

```
ERROR: set_data: Condition "ogg_stream == __null" is true.
   At: modules/stb_vorbis/audio_stream_ogg_vorbis.cpp:202
ERROR: import: Condition "!ogg_stream->get_data().size()" is true. Returned: ERR_FILE_CORRUPT
   At: modules/stb_vorbis/resource_importer_ogg_vorbis.cpp:102
ERROR: _reimport_file: Error importing 'res://hello.ogg'.
   At: editor/editor_file_system.cpp:1795
```

And the UI shows it failed to import:

![image](https://user-images.githubusercontent.com/331300/109398240-4a18ce80-7901-11eb-96de-58e163c9a6db.png)

The file plays fine in various players and opens in Audacity.

**Steps to reproduce:**

Create a new project and add `hello.ogg`: [hello.zip](https://github.com/godotengine/godot/files/6055248/hello.zip)

**Minimal reproduction project:**
[ogg_project.zip](https://github.com/godotengine/godot/files/6055254/ogg_project.zip)

**Comment by Calinou** [comment]
Is there a minimal `stb_vorbis` loader we can try out to see if this is an issue with stb_vorbis itself?

**Comment by Ev1lbl0w** [comment]
That's because `a.ogg` is actually a MP3 file :slightly_smiling_face: 

```
$ file a.ogg               
a.ogg: MPEG ADTS, layer III, v2,  32 kbps, 24 kHz, Monaural
```

**Comment by 31** [comment]
Oh... oops. 😄 Thanks. I don't know an easy way to tell on Windows (but I don't work with audio files much).

**Comment by Calinou** [comment]
> Oh... oops. smile Thanks. I don't know an easy way to tell on Windows (but I don't work with audio files much).

You can install FFmpeg and use the `ffprobe <path to file>` command, or install a port of `file` for Windows and use `file <path to file>`.

Both can be installed using [Scoop](https://scoop.sh):

```
scoop install ffmpeg file
```

**Comment by Ev1lbl0w** [comment]
I've noticed that [image_loader.cpp](https://github.com/godotengine/godot/blob/master/core/io/image_loader.cpp#L125) checks the header of it's resource when loading, to ensure the format is valid; maybe this could be implemented for all binary resources?

https://github.com/godotengine/godot/blob/8fbe644b6e6548dc2c9f9d675bc17f13b74e747b/core/io/image_loader.cpp#L137-L144

**Comment by akien-mga** [comment]
Fixed by #56903.
